File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/SmtpForm.ts
import { StringEditor, IntegerEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface SmtpForm {
    Server: StringEditor;
    Port: IntegerEditor;
    Username: StringEditor;
    Password: StringEditor;
    IsSsl: BooleanEditor;
}

export class SmtpForm extends PrefixedContext {
    static readonly formKey = 'Settings.Smtp';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SmtpForm.init)  {
            SmtpForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = BooleanEditor;

            initFormType(SmtpForm, [
                'Server', w0,
                'Port', w1,
                'Username', w0,
                'Password', w0,
                'IsSsl', w2
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Settings/Smtp/SmtpForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Settings.Forms;

[FormScript("Settings.Smtp")]
[BasedOnRow(typeof(SmtpRow), CheckNames = true)]
public class SmtpForm
{
    public string Server { get; set; }
    public int Port { get; set; }
    public string Username { get; set; }
    public string Password { get; set; }
    public bool IsSsl { get; set; }
}